### ISSUE TITLE
Pipeline initialization fails with MongoHistory and language.OpenNLP

### ISSUE THREAD

**Issue by jbaker-dstl** [dstl/baleen#30]
Trying to create the following pipeline causes the pipeline initialization to fail on language.OpenNLP.

```
history:
   class: uk.gov.dstl.baleen.history.mongo.MongoHistory

collectionreader:
  class: FolderReader
  folders:
  - corpus

annotators:
- language.OpenNLP
- class: gazetteer.Mongo
  collection: person_gazetteer
  valueField: name
  type: Person
- class: stats.OpenNLP
  model: en-ner-person.bin
  type: Person

consumers:
- Mongo
```

**Comment by jbaker-dstl** [comment]
Problem isn't specific to language.OpenNLP - MongoHistory appears to fail on the first annotator regardless.

**Comment by jbaker-dstl** [comment]
Problem is a change made in 2.2.0 where we cast the parameters of some configuration information into Strings, and thereby lose the resource that MongoHistory is dependent on (i.e. SharedMongoResource). Looking into the best way to fix this, and it will be addressed in the next release.

**Comment by jbaker-dstl** [comment]
Fixed in 2.3.0-SNAPSHOT
